connectors/slack/template/storage.js

```diff
@@ -1,47 +1,69 @@
 const Sdk = require('@fusebit/add-on-sdk');
 
-let storage;
-module.exports.ensureStorage = (ctx) => {
-  const twoMinutesFromNow = Date.now() + 2 * 60 * 1000;
-  if (!storage || storage.expiry < twoMinutesFromNow) {
-    storage = Sdk.getStorageClient(ctx);
-  }
-};
-
 let mappingTimestamp;
-module.exports.get = async () => {
+let etag;
+
+async function getStorageClient(ctx) {
+  return await Sdk.createStorageClient(
+    ctx,
+    ctx.fusebit.functionAccessToken,
+    `boundary/${ctx.boundaryId}/function/${ctx.functionId}`
+  );
+}
+
+module.exports.get = async (ctx) => {
+  const storage = await getStorageClient(ctx);
   let result = await storage.get();
-  module.exports.teamToHandler = (result && result.teamToHandler) || {};
-  module.exports.tokenToTeam = (result && result.tokenToTeam) || {};
-  mappingTimestamp = Date.now();
+  if (result && result.data && result.data.teamToHandler) {
+    module.exports.teamToHandler = result.data.teamToHandler;
+    mappingTimestamp = Date.now();
+    etag = result.etag;
+  }
 };
 
-module.exports.put = async (applyChanges) => {
+module.exports.put = async (ctx, applyChanges) => {
+  const storage = await getStorageClient(ctx);
+
   while (true) {
     // TODO Maybe add some backoff?
 
-    applyChanges();
+    if (typeof applyChanges === 'function') {
+      applyChanges();
+    }
 
     try {
-      await storage.put({
-        teamToHandler: module.exports.teamToHandler,
-        tokenToTeam: module.exports.tokenToTeam,
-      });
+      const payload = {
+        data: {
+          teamToHandler: module.exports.teamToHandler,
+        },
+      };
+      if (etag) {
+        payload.etag = etag;
+      }
+      const response = await storage.put(payload);
+      etag = response.etag;
       break;
     } catch (e) {
-      Sdk.debug('Cache conflict');
       if (e.statusCode && e.statusCode == 409) {
+        Sdk.debug('Cache conflict');
+
         // Local copy is stale, need to refresh
         await module.exports.get();
       } else throw e;
```

Comment:
Style; {} needed.  Has prettier been run on this file?

Reply:
Added and yes